### DISCUSSION TITLE
Theoretical use-case: offline, local-only Garmin data analysis?

### DISCUSSION THREAD

**Discussion by alexjmccoy** [tcgoetz/GarminDB#156] [General]
As someone focused on privacy and having my data local-only whenever possible, I'm intrigued by this project! I've enjoyed reading through the wiki and I'm looking forward to getting started with it.

My best-case-scenario use case would be the ability to analyze my Garmin data completely locally without ever uploading anything to Garmin's servers. Is such a thing possible with this software and Garmin Express? I believe much of the FirstBeat analytics (Body Battery, Training Readiness, etc) happens on the watch, and of course .FIT files may be analyzed locally.

I just sold my 945 LTE and I plan on soon deleting my entire data history from Garmin Connect. I am buying an Epix 2 on Friday. Is it theoretically possible to never register the new watch with Garmin Connect and use it entirely offline? I can update the watch with Garmin Express (or potentially Garmin Connect? Not sure what the app can do without an account or uploading to Garmin's servers).

If GarminDB allows it, I could then download all of my health and fitness data from the watch daily and analyze it in GarminDB. Is this approach doable with GarminDB? Does anyone else avoid using the Connect web service and focus on local-only analysis?

Thanks!

**Comment by tcgoetz** [comment]
It's possible. The hurdle is getting the files off locally. Older Garmin watches mount as USB mass storage device and you can copy the FIT files off directly. More recent Garmin watches use the MTP protocol. You need a MTP client to mount the watch and get access to the FIT files. Windows has MTP build in. On MacOS or Linux you need MTP client software.

Once you have the watch mounted and access to the FIT files on the watch, look at the copy (vs download) options of GarminDb to copy the FIT files into the GarminDB folders. After that step its the same as using GarminDb in download mode.

Let me know if you run into any issues. I haven't used copy mode since I got a Fenix 6 with MTP.

**Comment by alexjmccoy** [comment]
Thanks for the response! My understanding is that some health parameters such as sleep logging are processed server-side. Do you have a sense of how many of these Garmin metrics are done by Garmin on their servers vs. what happens on the watch?

Do you have a preferred MTP client for macOS?

**Comment by tcgoetz** [comment]
The FIT file contains sleep level by time, oxygen levels if the watch supports it, breathing rates, and heart rate. The online info should be able to be reconstructed.

There are a few MTP clients for MacOs, like the android one and OpenMTP. I can't speak on their use with a Garmin watch. If you work through it, please submit a howto article to the wiki.

**Comment by arpanghosh8453** [comment]
I was building a new app for offline activity analysis dashboard where you can process, store and compare activities recorded in the FIT files. 

The project is free and open source on Github : https://github.com/arpanghosh8453/fit-dashboard

Build a website for this as well :) 

[https://fitdashboard.app](https://fitdashboard.app)
